packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -253,32 +261,39 @@ class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
     return skiaClient.imgtestAdd(golden.path, goldenFile);
   }
 
-  /// Decides based on the current environment whether goldens tests should be
-  /// performed against Skia Gold.
+  /// Decides based on the current environment if goldens tests should be
+  /// executed through Skia Gold.
   static bool isAvailableForEnvironment(Platform platform) {
     final String cirrusPR = platform.environment['CIRRUS_PR'] ?? '';
     final String cirrusBranch = platform.environment['CIRRUS_BRANCH'] ?? '';
-    return platform.environment.containsKey('CIRRUS_CI')
+    final bool cirrusPostSubmit = platform.environment.containsKey('CIRRUS_CI')
       && cirrusPR.isEmpty
       && cirrusBranch == 'master'
       && platform.environment.containsKey('GOLD_SERVICE_ACCOUNT');
+
+    final bool luciPostSubmit = platform.environment.containsKey('SWARMING_TASK_ID')
+      && !platform.environment.containsKey('GOLD_TRYJOB');
```

Comment:
Is the containsKey('GOLD_TRYJOB') negated because this only runs on the waterfall?

Reply:
Indeed!